src/twisted/web/http.py

```diff
@@ -1205,6 +1206,16 @@ def write(self, data):
 
             for name, values in self.responseHeaders.getAllRawHeaders():
                 for value in values:
+                    if (
+                        self.method == b"CONNECT"
+                        and self.code // 100 == 2
+                        and name == b"Content-Length"
+                    ):
+                        self._log.info(
+                            "Warning: Removing Content-Length"
+                            " header in response to CONNECT request"
```

Comment:
Should we log the request here, or its URL?  Anything useful to include in the format string here?